tritium-slf4j/src/main/java/com/palantir/tritium/event/log/LoggingInvocationEventHandler.java

```diff
@@ -111,7 +109,8 @@ private void logInvocation(Method method, @Nullable Object[] nullableArgs, long
     }
 
     // All message formats are generated with placeholders and safe args
-    @SuppressWarnings({"Slf4jConstantLogMessage", "Slf4jLogsafeArgs", "Var"})
+    // switch generates larger bytecode and is less JIT inline friendly
+    @SuppressWarnings({"UseEnumSwitch", "Slf4jConstantLogMessage", "Slf4jLogsafeArgs", "Var"})
```

Comment:
I wonder how this would compare to an approach involving visitors? conversion from LoggingLevel to a loggingFunction (singleton for each level) which takes a format message and args. Maybe a few additional pointer jumps.

No change requested, just typing my thoughts

Reply:
yeah, I can test that at some point when I have cycles or if someone else wants to take a look at the performance characteristics across a few different JDK versions